FRW/Services/Manager/ProfileManager/ProfileManager.swift

```diff
@@ -0,0 +1,265 @@
+      let key = profile.uid + (profile.wallets.first?.address ?? "")
+      if let existingProfile = profileList[key], existingProfile.wallets.count > profile.wallets.count {
+        profileList[key] = existingProfile
+      } else {
+        profileList[key] = profile
+      }
+    }
+    return profileList.map { $0.value }
+  }
+}
+
+// MARK: Valid Profile List
+extension ProfileManager {
+  static func fetchValidUserId() -> [String] {
+    var userIdAndPublicKeyPre: [String] = []
+    // Secure Enclave Key
+    let seKeylist = SecureEnclaveKey.KeychainStorage.allKeys
+    guard let message = "test message".data(using: .utf8) else {
+      log.error("[Profile] encode message failed. This shouldn't happen")
+      return []
+    }
+
+    for key in seKeylist {
+      guard let se = try? SecureEnclaveKey.wallet(id: key) else {
+        log.warning("[Profile] SecureEnclaveKey get failed.\(key) ")
+        continue
+      }
+      guard let signature = try? se.sign(data: message, hashAlgo: .SHA2_256) else {
+        log.warning("[Profile] SecureEnclaveKey sign message failed.\(key) ")
+        continue
+      }
+      let result = se.isValidSignature(signature: signature, message: message)
+      if result {
+        userIdAndPublicKeyPre.append(key)
+      } else {
+        log.warning("[Profile] SecureEnclaveKey valid signature failed.\(key) ")
+      }
+    }
+    // SeedPhraseKey
+    let spKeyList = SeedPhraseKey.seedPhraseStorage.allKeys
+    for key in spKeyList {
+      guard let provider = try? SeedPhraseKey.wallet(id: key) else {
+        log.warning("[Profile] SeedPhraseKey get failed.\(key) ")
+        continue
+      }
+      var result = false
+      if let signature = try? provider.sign(data: message, signAlgo: .ECDSA_SECP256k1, hashAlgo: .SHA2_256) {
+        result = provider.isValidSignature(signature: signature, message: message, signAlgo: .ECDSA_SECP256k1)
+      }
+      
+      if !result,  let signature = try? provider.sign(data: message, signAlgo: .ECDSA_P256, hashAlgo: .SHA2_256) {
+        result = provider.isValidSignature(signature: signature, message: message, signAlgo: .ECDSA_P256)
+      }
+      if result {
+        userIdAndPublicKeyPre.append(key)
+      } else {
+        log.warning("[Profile] SeedPhraseKey valid signature failed.\(key) ")
+      }
+    }
+    // PrivateKey
+    let pkKeyList = FlowWalletKit.PrivateKey.PKStorage.allKeys
+    for key in pkKeyList {
+      guard let provider = try? FlowWalletKit.PrivateKey.wallet(id: key) else {
+        log.warning("[Profile] PrivateKey get failed.\(key) ")
+        continue
+      }
+      var result = false
+      if let signature = try? provider.sign(data: message, signAlgo: .ECDSA_P256, hashAlgo: .SHA2_256) {
+        result = provider.isValidSignature(signature: signature, message: message, signAlgo: .ECDSA_P256)
+      }
+      if !result, let signature = try? provider.sign(data: message, signAlgo: .ECDSA_SECP256k1, hashAlgo: .SHA2_256) {
```

Comment:
For private key type, I think we don't need test for sign ? Cause we can access the raw private key directly, right? 
Only test the Secure Enclave one, cause only when it signs it will know if we have the access or not